### PR TITLE
Change transaction section

### PR DESCRIPTION
**Pull request by neewy**
In this section I have added explanation for transaction statuses, and expanded the explanation.

Please check it with ruby and bundle gem installed on your machine:
https://github.com/lord/slate/wiki/Installing-Slate#getting-set-up

### CODE REVIEW

**Review by l4l — COMMENTED** [review]
Maybe add in advance MST_EXPIRED status?

File: source/includes/_transactions.md [hunk 1]
```diff
@@ -19,10 +19,17 @@ message Transaction {
 }
 ```
 
+Transaction is a state-chaging set of actions in the system. When a transaction passes validation and consensus stages, it is written in block and saved in immutable block store (block chain).
```
Comment (luckychess) [inline source/includes/_transactions.md]: chaging -> changing

File: source/includes/_transactions.md [hunk 2]
```diff
@@ -19,10 +19,17 @@ message Transaction {
...
+Transactions consist of commands, performing an action over an entity in the system. The entity might be an account, asset, etc. — more in [entity-relationship model](#er-model) page. 
+
+<aside class="notice">
+Iroha API follows command-query separation principle, which is described <a href="https://en.wikipedia.org/wiki/Command–query_separation">here</a>.
+</aside>
 
-Client application should follow descibed protocol and form transactions accordingly to description below.
+
+Communication between Iroha peer and client application is maintained via [gRPC framework](https://grpc.io/about/). 
+Client applications should follow descibed protocol and form transactions accordingly to description below.
```
Comment (luckychess) [inline source/includes/_transactions.md]: descibed -> described

File: source/includes/_transactions.md [hunk 3]
```diff
@@ -77,18 +82,32 @@ message Signature {
...
 
-**Payload** stores everything except signatures: 
+**Payload** stores all transaction fields, except signatures: 
 <ul>
     <li> Time of creation (unix time, in milliseconds) </li> 
-    <li> ID of transaction creator (username@domain) </li>
-    <li> Transaction counter — counts how many times transaction creator send transactions in total. Counter is used to prevent replays, and is formed on client side. </li>
-    <li> Repeated commands which would be described in details later</li> 
+    <li> Account ID of transaction creator (username@domain) </li>
+    <li> Transaction counter. It counts how many times transaction creator send transactions in total. Counter is used to prevent replay attack, and is formed on client side </li>
```
Comment (luckychess) [inline source/includes/_transactions.md]: sent I think

File: source/includes/_transactions.md [hunk 4]
```diff
@@ -77,18 +82,32 @@ message Signature {
...
 <aside class="notice">
-In preview version, there is no way to get current transaction counter by any means. You have to remember it on a client side and increment it. In future version this will be changed, and client application may retrieve this information.
+In current version, there is no way to get current transaction counter by any means. You have to remember it on a client side and increment it. In future version this will be changed, and client application may retrieve this information.
 </aside>
 
 ## Transaction statuses
+
+Current version of Iroha peer follows client [pull principle](https://en.wikipedia.org/wiki/Pull_technology) for networking. It means that client should be proactive and request the state of transaction from peer. 
+
+This section describes the set of states and mathes them with transaction lifecycle.
```
Comment (luckychess) [inline source/includes/_transactions.md]: matches?

File: source/includes/_transactions.md [hunk 5]
```diff
@@ -77,18 +82,32 @@ message Signature {
...
+
+![Block](../images/tx_status.png "Block structure")
+
+ * `NOT_RECEIVED`: as client can query the peer, which may not have this transaction, the response indicates about such situation.
```
Comment (luckychess) [inline source/includes/_transactions.md]: As far as I know, if peer does not know about transaction means there is no such transaction at all.

**Review by luckychess — CHANGES_REQUESTED** [review]

File: source/includes/_transactions.md (hunk 4 above)
Comment (kamilsa) [reply source/includes/_transactions.md]: correct, no?

**Review by kamilsa — CHANGES_REQUESTED** [review]

**Comment by neewy** [comment]
Next time I will use grammarly before opening PR, thanks

**Comment by neewy** [comment]
> Maybe add in advance MST_EXPIRED status?

@l4l I think that we should update docs in future as soon as we are merging the change in the repo

File: source/includes/_transactions.md [hunk 6]
```diff
@@ -86,28 +86,28 @@ message Signature {
...
 
-This section describes the set of states and mathes them with transaction lifecycle.
+This section describes the set of states and matches them with transaction lifecycle.
 
 ![Block](../images/tx_status.png "Block structure")
 
- * `NOT_RECEIVED`: as client can query the peer, which may not have this transaction, the response indicates about such situation.
- * `STATELESS_VALIDATION_FAILED`: the transaction was formed with some fields, not meeting constraints. This status is returned to client, who formed transaction, right after the transaction was sent. 
- * `STATELESS_VALIDATION_SUCCESS`: the transaction has successfully passed stateless validation. This status is returned to client, who formed transaction, right after the transaction was sent. 
+ * `NOT_RECEIVED`: the peer does not this transaction.
```
Comment (luckychess) [inline source/includes/_transactions.md]: Verb missed (know about or something else)

**Review by luckychess — APPROVED** [review]

File: source/includes/_transactions.md (hunk 6 above)
Comment (neewy) [reply source/includes/_transactions.md]: sorry

File: source/includes/_transactions.md [hunk 7]
```diff
@@ -77,18 +82,32 @@ message Signature {
...
 
 ## Transaction statuses
+
+The current version of Iroha peer follows client [pull principle](https://en.wikipedia.org/wiki/Pull_technology) for networking. It means that client should be proactive and request the state of a transaction from a peer. 
+
+This section describes the set of states and matches them with transaction lifecycle.
+
+![Block](../images/tx_status.png "Block structure")
+
+ * `NOT_RECEIVED`: the peer does not this transaction.
```
Comment (kamilsa) [inline source/includes/_transactions.md]: does not what?

File: source/includes/_transactions.md [hunk 8]
```diff
@@ -77,18 +82,32 @@ message Signature {
...
+ * `STATELESS_VALIDATION_FAILED`: the transaction was formed with some fields, not meeting constraints. This status is returned to a client, who formed transaction, right after the transaction was sent. 
+ * `STATELESS_VALIDATION_SUCCESS`: the transaction has successfully passed stateless validation. This status is returned to a client, who formed transaction, right after the transaction was sent. 
+ * `STATEFUL_VALIDATION_FAILED`: the transaction has commands, which violate validation rules, checking state of the chain (e.g. asset balance, account permissions, etc.) 
+ * `STATEFUL_VALIDATION_SUCCESS`: the transaction has successfully passed stateful validation.
+ * `IN_PROGRESS`: the transaction is the part of a block, which is on voting during current consensus round.
```
Comment (kamilsa) [inline source/includes/_transactions.md]: If I am not mistaken, IN_PROGRESS means that transaction was received, but not processed.

**Review by kamilsa — CHANGES_REQUESTED** [review]

**Review by kamilsa — APPROVED** [review]

**Pull request merged**